File: C++/GLCM.cpp

#include "Metodos.hpp"
#include <iostream>

vector<double> glcm_extraction(Mat image, int grayscale){
    vector<double> glcm_features;
    
    Mat imageQuantized, coOccurence, coOccurenceNormalized;
    image.copyTo(imageQuantized);
    
    /* Quantiza a imagem de entrada */
    //quantizerMatrix(image, imageQuantized, grayscale);
    
    /* Calcula a Matriz de Co-Ocorrencia */
    getCoOccurrenceMatrix(imageQuantized, coOccurence, grayscale);
    
    /* Normaliza a Matriz de Co-Ocorrencia */
    normalizeCoOccurrenceMatrix(coOccurence, coOccurenceNormalized, imageQuantized, grayscale);
    
    /* Calcula os atributos da GLCM */
    getFeatures(glcm_features, coOccurenceNormalized, grayscale);
    
    imageQuantized.release();
    coOccurence.release();
    coOccurenceNormalized.release();
    
    return glcm_features;
}
int main(){
	Mat imageMethods = imread("c1_1.jpg",CV_LOAD_IMAGE_GRAYSCALE);
	vector<double> data;
	data = glcm_extraction(imageMethods, 256);
	printf("\nData: %f",data[0]);
}

File: Python/GLCM.py
from Metodos import *

# Load an color image in grayscale
img = cv2.imread('../../couro_images_semfundo/c1_1.JPG',0)
imgQuantized = img.copy()

coOccurence = getCoOccurrenceMatrixMod(imgQuantized, 256)
coOccurenceNormalized = normalizeCoOccurrenceMatrix(coOccurence,imgQuantized,256)
glcm_features  = getFeatures(coOccurenceNormalized, 256)


print glcm_features
